apps/OpenSpace/ext/launcher/include/sgctedit/settingswidget.h

```diff
@@ -72,17 +74,81 @@ Q_OBJECT
+     * windows is 2 or more. 
+     */
+    void setEnableShowUiOnFirstWindowCheckbox(bool enable);
+
+    /**
+     * Gets the value of the selection for which display first window should mirror if
+     * the option to show the Ui in the first window is enabled. Note that this will
+     * return a value even if the checkbox is not enabled.
+     * 
+     * \return -1 if in a disabled state (e.g. when showUiOnFirstWindow() returns false)
+     *          0 if no window is being mirrored (only the UI will appear in first
+     *            window, with no 3D rendering)
+     *          1 if the second window's viewport is mirrored in the first window
+     *          2 if the third window's viewport is mirrored in the first window
+     *          3 if the fourth window's viewport is mirrored in the first window
+     */
+    int graphicsSelectionForShowUiOnFirstWindow() const;
+
+    /**
+     * Sets value of the graphics selection combo box for which other window that the
+     * first window will mirror.
+     * 
+     * \param selection int value for the combo box selection.
+     *                  0 if no window is being mirrored (only the UI will appear)
+     *                  1 if the second window's viewport is mirrored in the first window
+     *                  2 if the third window's viewport is mirrored in the first window
+     *                  3 if the fourth window's viewport is mirrored in the first window
+     */
+    void setGraphicsSelectionForShowUiOnFirstWindow(int selection);
+
     /**
      * Sets the value of the checkbox for enabling VSync.
      * 
      * \param enableVsync boolean value, if set true then VSync is enabled
      */
     void setVsync(bool enableVsync);
 
+    /**
+     * Called when the number of windows that should be displayed changes.
+     * 
+     * \param newCount The new number of windows included
+     */
+    void nWindowsDisplayedChanged(int newCount);
+
+    /**
+     * Gets the pointer to the QComboBox that selects the graphics for first window
+     * 
+     * \return pointer to the QComboBox object
+     */
+    QComboBox* firstWindowGraphicsSelection();
+
+    /**
+     * Gets the pointer to the QCheckBox that selects if UI is in first window only
+     * 
+     * \return pointer to the QCheckBox object
+     */
+    QCheckBox* showUiOnFirstWindowCheckbox();
+
+signals:
+    void firstWindowGraphicsSelected(int selection);
+
 private:
     sgct::quat _orientationValue = sgct::quat(0.f, 0.f, 0.f, 0.f);
     QCheckBox* _checkBoxVsync = nullptr;
     QCheckBox* _showUiOnFirstWindow = nullptr;
+    QComboBox* _firstWindowGraphicsSelection = nullptr;
+    QBoxLayout* _firstWindowSelectionLayout = nullptr;
+    int _stateOfUiOnFirstWindowPreviousCount = 1;
+    bool _stateOfUiOnFirstWindowWhenDisabled = false;
+
+    void showUiOnFirstWindowClicked(bool checked);
+    void firstWindowGraphicsSelectionChanged(const QString &text);
```

Comment:
We place the function declarations as the first thing in the visibility block, so these two functions would come directly after the `private` and before the member variables